### PR TITLE
support string bytes

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: test/fixture/stringbytes.scrypt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+library L {
+  static const bytes p = "😊"; // emoji
```
Comment (xhliu) [inline test/fixture/stringbytes.scrypt:3]: add a test for empty string ""

File: test/stringbytes.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import { bsv, toHex, getPreimage, toHashedMap } from '../src/utils';
+import { Bytes, SigHash, SigHashPreimage } from '../src';
+import { String } from '../src/scryptTypes';
+const inputIndex = 0;
+const inputSatoshis = 100000;
+
+const outputAmount = inputSatoshis
+
+
+describe('stringbytes.test', () => {
```
Comment (xhliu) [inline test/stringbytes.test.ts:14]: test for empty string/bytes

File: test/stringbytes.test.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    let instance, result;
+
+    const Test = buildContractClass(loadDescription('stringbytes1_desc.json'));
+    const { L } = buildTypeClasses(Test);
+    before(() => {
+      instance = new Test(new L(new Bytes("1234ab"), new String("你好world"), new String("abcd"), new String("こんにちは"), new String("b'aa'"), new String("😊")),
+        new Bytes("1234ab"), new String("你好world"), new String("abcd"), new String("こんにちは"), new String("b'aa'"), new String("😊"));
+    });
+
+    it('should success when call unlock', () => {
```
Comment (xhliu) [inline test/stringbytes.test.ts:56]: should succeeding when calling unlock


Change everywhere

File: test/toliteral.test.ts [hunk 4]
```diff
@@ -137,4 +138,20 @@ describe('toLiteral test', () => {
         })
     })
 
+
+    describe('test UTF8Bytes', () => {
+
+        it('should show right', () => {
```
Comment (xhliu) [inline test/toliteral.test.ts:144]: Also test empty string
